diff --git a/src/flash/nor/stm32f1x.c b/src/flash/nor/stm32f1x.c
index 8e66af368..9769e4b3d 100644
--- a/src/flash/nor/stm32f1x.c
+++ b/src/flash/nor/stm32f1x.c
@@ -837,6 +837,12 @@ static int stm32x_probe(struct flash_bank *bank)
 		stm32x_info->default_rdp = 0xAA;
 		stm32x_info->can_load_options = true;
 		break;
+	case 0x346:
+	case 0x347:
+		page_size = 512;
+		stm32x_info->ppage_size = 8;
+		max_flash_size_in_kb = 1024;
+		break;
 	case 0x446: /* stm32f303xD/E */
 		page_size = 2048;
 		stm32x_info->ppage_size = 2;
